WCOSS2 run support and script/config updates

KateFriedman-NOAA opened the pull request:
**Description**

This PR includes script and run support updates for running the `develop` branch on WCOSS2. This PR does not include resource updates for running on WCOSS2 and does not complete the port to WCOSS2. Additional WCOSS2 `develop` port PRs will follow this PR.

Changes in this PR mostly address the following:

1. WCOSS2 port changes requested by NCO (e.g. `NWPROD`/`NWROOT` -> `PACKAGEROOT`) in the move of operations to WCOSS2 in the past year.
2. Adding WCOSS2 support in the way of creating `WCOSS2.env`, a `wcoss2.yaml` hosts file for rocoto, and config updates.
3. Some cleanup that was also done in the operations GFS port to WCOSS2.

More specifically, the changes in this PR are:

1. Create `WCOSS2.env` under the env folder.
2. Remove references/usage of jlogfile thoughout.
3. Change `NWPROD` to `PACKAGEROOT` throughout.
4. Replace `NWROOT` with `OPSROOT` where appropriate.
5. Env file updates (non-WCOSS2):
  * Update "anal" if block in `HERA.env` and `ORION.env` to add "or" for "analcalc"; needed to match similar setting in `WCOSS2.env` and change in `jobs/JGLOBAL_ATMOS_ANALYSIS_CALC`
6. Update JJOB scripts:
  * Update `jobs/JGLOBAL_ATMOS_ANALYSIS_CALC` to pass "analcalc" to env file instead of "anal"
  * Move config and env file sourcing from rocoto job scripts into JJOBs for gempak
  * Update the default COM* paths to utilize `compath.py` and the new `compath.py` format; includes updating `COMINukmet` and `COMINecmwf` defaults to new `compath.py` format
  * Add `COMINobsproc` to `jobs/JGFS_ATMOS_GEMPAK_NCDC_UPAPGIF`
  * Replace `COMINgfs` with `COMINobsproc` and replace `COMINgfs_m6hrs` with `COMIN_m6hrs` in `jobs/JGLOBAL_ATMOS_EMCSFC_SFC_PREP`
  * Update `ARCHSYND`, `HOMENHC` and `TANK_TROPCY` in `jobs/JGLOBAL_ATMOS_TROPCY_QC_RELOC`, remove `HOMENHCp1`
  * Update default `APRUN` commands for several downstream jobs (`mpirun` -> `mpiexec`)
  * Set `wavempexec` to `wavempexec` instead of `launcher` in `jobs/JGLOBAL_WAVE_INIT`
  * Correct list of configs to source in `jobs/JGLOBAL_WAVE_POST_BNDPNTBLL` to include `wavepostbndpntbll` instead `wavepostbndpnt`
  * Update how `PDY` is invoked (`. PDY` to `. ./PDY`)
  * Update the `RUN` and `RUN2` variables in `jobs/JGFS_ATMOS_GEMPAK_PGRB2_SPEC`
  * Some cleanup to remove unneeded `RUN_ENVIR=nco` checks and some extra whitespaces.
7. Update rocoto job scripts:
  * Move config and env sourcing out of gempak rocoto job scripts (moved to JJOB scripts); also change to invoke JJOB directly and add `exit $status`
  * Update fit2obs and tracker section of `jobs/rocoto/vrfy.sh`
  * Add missing env file sourcing to `jobs/rocoto/wafsgcip.sh`
8. Update `module_base.wcoss2.lua` to use para version of `ncdiag` module (update to use official ops install later)
9. Updates to configs:
  * Whitespace cleanup in `config.anal`
  * Add `export nth_echgres=$nth_echgres_gfs` when CDUMP=gfs in `config.analcalc`
  * Add `COMINsyn` to `config.base.emc.dyn`
  * Update `ROTDIR` in `config.base.nco.static`
  * Change `OUTPUT_FILETYPE_` variables to be "netcdf" on all machines in `config.efcs`
  * Updates to `config.fv3.nco.static`: remove non-WCOSS2 references, add `npe_node_max=128` for WCOSS2, change C384 DELTIM to 200 (ops value), and update variables for C384/C768 to ops values
  * Remove `GEMPAKSH=$HOMEgfs/jobs/JGFS_ATMOS_GEMPAK` line in `config.gempak`; now invoking `$HOMEgfs/jobs/JGFS_ATMOS_GEMPAK` directly in the JJOB
  * Update `COMINsyn` default in `config.prep` and `config.vrfy`
  * Create `config.resources.nco.static`, copy from `operations`; will need updating for GFSv17 opd
  * Tracker updates in `config.vrfy`
10. Updates to scripts:
  * Updates to `RUN` and `RUN2` variables in `scripts/exgdas_atmos_nawips.sh`, `scripts/exgfs_atmos_goes_nawips.sh`, and `scripts/exgfs_atmos_nawips.sh`
  * Replace `COMINgfs` with `COMINobsproc` in `scripts/exgfs_atmos_gempak_gif_ncdc_skew_t.sh`
  * Update `APRUN` in `scripts/exgfs_atmos_gempak_meta.sh` (`mpirun` -> `mpiexec`)
11. Update `sorc/link_workflow.sh` to also copy the appropriate versions of `config.fv3` and `config.resources` in NCO mode.
12. Updates to ush scripts:
  * Remove references to `NWPROD` and replace with appropriate variables or paths.
  * Add echos to `ush/gfs_bufr.sh` to print out the specific forecast hour files for SPAs to check in case of failure. (Added by GDIT and Bo Cui)
  * Delete `ush/global_extrkr.sh` (not used anywhere anymore).
13. Update util source codes to build with `ftn` and include `-g -traceback` `FFLAGS`
14. Updates to setup script files:
  * Replace `nwprod` with `packageroot` in hosts files and `setup_expt.py`.
  * Create `wcoss2.yaml` host file.
  * Update `workflow_tasks.py` to set native xml tag for pbspro.

Issue #419

**Type of change**

- [x] New feature (non-breaking change which adds functionality)

**How Has This Been Tested?**

- [x] Clone and build tests on WCOSS2, Hera, and Orion
- [x] Cycled atmos-only tests on WCOSS2, Hera, and Orion (C192C96L127 on all machines and C768C384L127 on WCOSS2)
- [x] Forecast-only coupled tests on Hera and Orion

No downstream/ops-only jobs were invoked in testing, however updates to those job scripts are included in this PR.
  
**Checklist**

- [ ] My code follows the style guidelines of this project
- [x] I have performed a self-review of my own code
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] My changes need updates to the documentation. I have made corresponding changes to the documentation
- [x] My changes generate no new warnings
- [x] New and existing tests pass with my changes
- [ ] Any dependent changes have been merged and published

KateFriedman-NOAA committed:
Update repository name for EMC_post component to UPP

- the "EMC_post" repository was renamed to "UPP" on September 6th 2021
- update repository url in checkout script and Externals.cfg

Refs: #433

KateFriedman-NOAA committed:
Merge branch 'NOAA-EMC:develop' into develop

KateFriedman-NOAA committed:
Merge branch 'NOAA-EMC:develop' into develop

KateFriedman-NOAA committed:
Merge branch 'NOAA-EMC:develop' into develop

KateFriedman-NOAA committed:
Merge branch 'NOAA-EMC:develop' into develop

KateFriedman-NOAA committed:
Merge branch 'NOAA-EMC:develop' into develop

KateFriedman-NOAA committed:
Merge branch 'NOAA-EMC:develop' into develop

KateFriedman-NOAA committed:
Merge branch 'NOAA-EMC:develop' into develop

KateFriedman-NOAA committed:
Merge branch 'NOAA-EMC:develop' into develop

KateFriedman-NOAA committed:
Merge branch 'NOAA-EMC:develop' into develop

KateFriedman-NOAA committed:
Merge branch 'NOAA-EMC:develop' into develop

KateFriedman-NOAA committed:
Merge branch 'NOAA-EMC:develop' into develop

KateFriedman-NOAA committed:
Merge branch 'NOAA-EMC:develop' into develop

KateFriedman-NOAA committed:
Merge branch 'NOAA-EMC:develop' into develop

KateFriedman-NOAA committed:
Merge branch 'NOAA-EMC:develop' into develop

KateFriedman-NOAA committed:
Merge branch 'NOAA-EMC:develop' into develop

KateFriedman-NOAA committed:
Merge branch 'NOAA-EMC:develop' into develop

KateFriedman-NOAA committed:
Merge branch 'NOAA-EMC:develop' into develop

KateFriedman-NOAA committed:
Merge branch 'NOAA-EMC:develop' into develop

KateFriedman-NOAA committed:
Merge branch 'NOAA-EMC:develop' into develop

KateFriedman-NOAA committed:
Merge branch 'NOAA-EMC:develop' into develop

KateFriedman-NOAA committed:
Merge branch 'NOAA-EMC:develop' into develop

KateFriedman-NOAA committed:
Add WCOSS2 operational GFS defs files

- commit GFSv16.2 WCOSS2 operational GFS defs files as of April 7th 2022
- final changes to defs files will be merged later

Refs: #399

KateFriedman-NOAA committed:
Add 06, 12, and 18 WCOSS2 operational GFS defs

Refs: #399

KateFriedman-NOAA committed:
Merge branch 'NOAA-EMC:develop' into develop

KateFriedman-NOAA committed:
Merge branch 'NOAA-EMC:develop' into develop

KateFriedman-NOAA committed:
Merge branch 'NOAA-EMC:develop' into develop

KateFriedman-NOAA committed:
Merge branch 'NOAA-EMC:develop' into develop

KateFriedman-NOAA committed:
Merge branch 'NOAA-EMC:develop' into develop

lgannoaa committed:
Update checkout.sh to remove git submodule update for upp checkout (#768)

REF:767

KateFriedman-NOAA committed:
Merge branch 'NOAA-EMC:develop' into develop

KateFriedman-NOAA committed:
Merge branch 'NOAA-EMC:develop' into develop

KateFriedman-NOAA committed:
Merge branch 'NOAA-EMC:develop' into develop

KateFriedman-NOAA committed:
Merge branch 'NOAA-EMC:develop' into develop

KateFriedman-NOAA committed:
Merge branch 'NOAA-EMC:develop' into develop

KateFriedman-NOAA committed:
Merge branch 'NOAA-EMC:develop' into develop

KateFriedman-NOAA committed:
Merge branch 'NOAA-EMC:develop' into develop

KateFriedman-NOAA committed:
Merge branch 'NOAA-EMC:develop' into develop

KateFriedman-NOAA committed:
Merge branch 'NOAA-EMC:develop' into develop

KateFriedman-NOAA committed:
Merge branch 'NOAA-EMC:develop' into develop

KateFriedman-NOAA committed:
Merge branch 'NOAA-EMC:develop' into develop

KateFriedman-NOAA committed:
Merge branch 'NOAA-EMC:develop' into develop

KateFriedman-NOAA committed:
Merge branch 'NOAA-EMC:develop' into develop

KateFriedman-NOAA committed:
Merge branch 'NOAA-EMC:develop' into develop

KateFriedman-NOAA committed:
Merge branch 'NOAA-EMC:develop' into develop

KateFriedman-NOAA committed:
Move transfer files into new transfer folder

- In move to WCOSS2 the transfer*list files were moved into a new
transfer folder under the upper-level parm folder.

Refs: #419

KateFriedman-NOAA committed:
Updated transfer list files from WCOSS2 ops

- The transfer*list files were updated in operations during the move
from WCOSS1 to WCOSS2.
- The beginning of the path within the transfer*list files was updated
from:
  com/gfs/_ENVIR_
to:
  _COMROOT_/gfs/_SHORTVER_

Refs: #419

KateFriedman-NOAA committed:
Add GFSv16.1 and GFSv16.2 release notes

- Add release notes for GFSv16.1 and GFSv16.2 versions that were added
into the operations branch but not yet in the develop branch.

Refs: #419

KateFriedman-NOAA committed:
Merge branch 'develop' into feature/dev-wcoss2

* develop:
  Updated GFS transfer*list files from operations (develop) (#918)

KateFriedman-NOAA committed:
WCOSS2 updates to gempak ush scripts

- Add /gempak subfolder where needed in gempak ush scripts.
- Remove unneeded commented out path settings from older iterations.

Refs: #419

KateFriedman-NOAA committed:
Removing older release notes

- Cleaning out older GFS version release notes.
- Includes current GFSv16.2.1 release notes, will commit GFSv16.3
release notes with implementation this fall.
- Will then keep only the latest release notes moving forward.

Refs: #419

KateFriedman-NOAA committed:
Merge branch 'develop' into feature/dev-wcoss2

* develop:
  Add postsnd job when bufrsnd it on (#926)

KateFriedman-NOAA committed:
Merge branch 'develop' into feature/dev-wcoss2

* develop:
  Add GDASapp (first wave of JEDI changes) (#871)

KateFriedman-NOAA committed:
Merge branch 'develop' into feature/dev-wcoss2

* develop:
  WCOSS2 gempak ush scripts updates and cleanup of old release notes (#920)

KateFriedman-NOAA committed:
Merge branch 'develop' into feature/dev-wcoss2

* develop:
  change gdasechgres dependency to just gdasefcs01 instead of gdasefmn (#933)
  Remove obsolete platforms (WCOSS1, Dell, Cray, Theia) references. (#922)
  Optimize DA clone, build, and link (#931)

KateFriedman-NOAA committed:
Merge branch 'NOAA-EMC:develop' into develop

KateFriedman-NOAA committed:
Merge branch 'develop' into feature/dev-wcoss2

* develop:
  Initial commit of directory comparison tools (#934)
  Add preamble, convert to bash, and remove env (#929)
  minimal intervention to create a data-atmosphere xml (#936)
  Remove Cray, Dell, WCOSS1 from module-setup.sh.inc (#943)
  bring GDASApp jjobs and exscripts to global-workflow (#941)

KateFriedman-NOAA committed:
Merge branch 'NOAA-EMC:develop' into develop

KateFriedman-NOAA committed:
Merge branch 'NOAA-EMC:develop' into develop

KateFriedman-NOAA committed:
Merge branch 'develop' into feature/dev-wcoss2

* develop:
  Remove SDATE=CDATE IAU block in NCO config.base (#963)
  Updates for P8  (#946)
  Fix GLDAS j-job link (#954)
  Add ocean post to archive dependencies (#949)

KateFriedman-NOAA committed:
Add WCOSS2 to module-setup.sh.inc

Refs: #419

KateFriedman-NOAA committed:
Add WCOSS2 to link_workflow.sh

Refs: #419

KateFriedman-NOAA committed:
Add WCOSS2 and module reset to machine-setup.sh

- Add a WCOSS2 block to machine-setup.sh.
- Change the Hera and Orion blocks to use "module reset" instead of
"module purge"; includes hack from RDHPCS sysadmins for using the
"module reset" command on those machines successfully.

Refs: #419

KateFriedman-NOAA committed:
Add WCOSS2 to load_fv3gfs_modules.sh

Refs: #419

KateFriedman-NOAA committed:
Create LUA module_base for WCOSS2

Refs: #419

KateFriedman-NOAA committed:
Create LUA modulefile for WW3 on WCOSS2

Refs: #419

KateFriedman-NOAA committed:
Create LUA modulefile for workflow_utils on WCOSS2

Refs: #419

KateFriedman-NOAA committed:
Merge branch 'develop' into feature/dev-wcoss2

* develop:
  Externals.cfg was stale and had drifted off (#965)
  Fix post comparison with zero-padded numbers (#964)

KateFriedman-NOAA committed:
Merge branch 'NOAA-EMC:develop' into develop

KateFriedman-NOAA committed:
Merge branch 'develop' into feature/dev-wcoss2

* develop:
  Add initial config files for global aerosol DA (#986)
  Update diag table to remove wav-ocn coupling fields (#979)
  use a robust Findwgrib2.cmake to find wgrib2 built w/ native wgrib2 build (#970)

KateFriedman-NOAA committed:
Add WCOSS2 to module-setup.csh.inc

Refs: #419

KateFriedman-NOAA committed:
Change WCOSS2 FIX_DIR to noscrub space

Refs: #419

KateFriedman-NOAA committed:
Change WCOSS2 machine detection path

- Change the path used to check if on WCOSS2 from "lfs/h2" to "/lfs/f1"

Refs: #419

KateFriedman-NOAA committed:
Merge branch 'NOAA-EMC:develop' into develop

KateFriedman-NOAA committed:
Merge branch 'develop' into feature/dev-wcoss2

* develop:
  Update UFS_UTILS tag to `ufs_utils_1_8_0` (#1001)
  Fix preamble id (#996)
  Add missing "atmos" into job dependencies (#998)
  Bugfix in arch.sh to remove hardwired "htar" (#992)
  Add in stubs for aerosol DA tasks + bugfix for setup_expt where cycled and ATMA are used (#990)
  Add GSI monitor scripts (#969)
  Fix product generation at some fcst hrs (#988)

KateFriedman-NOAA committed:
Revert module reset changes for R&D

- Pull out the changes to replace "module purge" with "module reset" for
the R&D machines in machine-setup.sh.
- Will commit separately after testing on R&Ds.
- Removing non-WCOSS2 changes from branch ahead of WCOSS2 build PR.

Refs: #419

KateFriedman-NOAA committed:
Update workflow PACKAGEROOT and create wcoss2 host

- Create new workflow/hosts/wcoss2.yaml file
- Update NWPROD->PACKAGEROOT variable in hera.yaml and orion.yaml
- Update NWPROD->PACKAGEROOT variable in setup_expt.py

Refs #419

KateFriedman-NOAA committed:
Add preliminary copy of config.resources.nco.static

Refs #419

KateFriedman-NOAA committed:
Add preliminary copy of WCOSS2.env

- Need to add new jobs to WCOSS2.env file still.

Refs #419

KateFriedman-NOAA committed:
Update native tag in workflow_tasks.py

- Update native tag in workflow_tasks.py to add PBSpro support.
- Add "debug=true" and "place=vscatter" into rocoto native tag in
generated xmls.
- Set to make job exclusive if "is_exclusive=True" is set.

Refs #419

KateFriedman-NOAA committed:
Update config copies in link_workflow.sh

- Update config copies at end of link_workflow.sh to also copy
config.fv3.nco.static to config.fv3 and copy config.resources.nco.static
to config.resources when $RUN_ENVIR = nco.

Refs #419

KateFriedman-NOAA committed:
Add para ncdiag module load to module_base.wcoss2.lua

- Similar to hack in GSI, load the para install of the ncdiag module on
WCOSS2 for use during runtime.

Refs #419

KateFriedman-NOAA committed:
Merge branch 'NOAA-EMC:develop' into develop

KateFriedman-NOAA committed:
Merge branch 'develop' into feature/dev-wcoss2

* develop:
  Add shell linter Github action for pull requests (#1007)
  Build updates for WCOSS2 (#1002)
  Update checkout.sh to remove git submodule update for upp checkout (#768)
  Add 06, 12, and 18 WCOSS2 operational GFS defs
  Add WCOSS2 operational GFS defs files
  Update repository name for EMC_post component to UPP

KateFriedman-NOAA committed:
Merge branch 'develop' into feature/dev-wcoss2

* develop:
  Correct shell linter config (#1013)
  Correct diagnostic file handling in ush/ozn_xtrct.sh (#1016)

KateFriedman-NOAA committed:
Merge branch 'NOAA-EMC:develop' into develop

KateFriedman-NOAA committed:
Folder WCOSS2 port changes into util files

- Change webtitle makefile to use ftn instead of ifort and add traceback
flags to build.
- Clean up ush/make_tif.sh.

Refs #419

KateFriedman-NOAA committed:
Folder ops WCOSS2 port changes into ush scripts

- Change NWPROD to PACKAGEROOT.
- Change NWROOT to OPSROOT.
- Retire jlogfile usage and references; includes removing postmsg.
- Add in some error checking for gfs_bufr job (from GDIT/Bo Cui).
- Some cleanup.

Refs #419

KateFriedman-NOAA committed:
Fold in ops WCOSS2 port changes into ex-scripts

- Retire jlogfile and postmsg usage and references.
- Change NWPROD to PACKAGEROOT.
- Change COMINgfs to COMINobsproc in gempak script.
- Change RUN to RUN2 in awips script.

Refs #419

KateFriedman-NOAA committed:
Add env file sourcing to wafsgcip rocoto job script

Refs #419

KateFriedman-NOAA committed:
Update vrfy rocoto job script

- Remove jlogfile references.
- Update fit2obs section for new version.
- Add COMINsyn setting with compath.py usage for tracker.

Refs #419

KateFriedman-NOAA committed:
Update JOBSPROC_GLOBAL_PREP path in rocoto prep.sh

- Change HOMEobsproc_network to HOMEobsproc.

Refs #419

KateFriedman-NOAA committed:
Remote jlogfile reference in awips rocoto script

Refs #419

KateFriedman-NOAA committed:
Update ocnpost rocoto job script

- Remove jlogfile reference.
- Change NWROOT to PACKAGEROOT.

Refs #419

KateFriedman-NOAA committed:
Update gempak rocoto job script

- Remove config and env sourcing, as well as some additional settings (moving to JJOB script).
- Add JJOB script invocation: $HOMEgfs/jobs/JGFS_ATMOS_GEMPAK
- Some cleanup.

Refs #419

KateFriedman-NOAA committed:
Change NWROOT to PACKAGEROOT in EFSOI job scripts

Refs #419

KateFriedman-NOAA committed:
Change NWROOT to PACKAGEROOT in ENKF job scripts

Refs #419

KateFriedman-NOAA committed:
WCOSS2 updates in wave job scripts

- Change NWROOT to PACKAGEROOT.
- Update PDY syntax.
- Update compath.py format.
- Some cleanup, including removal of RUN_ENVIR=nco blocks.
- Change wavempexec override to be prior setting of wavempexec.

Refs #419

KateFriedman-NOAA committed:
WCOSS2 updates for GEMPAK job scripts

- Update PDY syntax.
- Update compath.py format.
- Update APRUN default to use mpiexec launcher with needed arguments.
- Change NWROOT to PACKAGEROOT.
- Update DATA default.
- Add SENDCOM setting with default of "NO".
- Add config and env file sourcing, moved from rocoto job script level.

Refs #419

KateFriedman-NOAA committed:
WCOSS2 updates for TROPCY_QC_RELOC JJOB script

- Change NWROOT to PACKAGEROOT.
- Remove RUN_ENVIR=nco block.
- Update ARCHSYND, HOMENHC, and TANK_TROPCY settings.

Refs #419

KateFriedman-NOAA committed:
WCOSS2 updates to GEMPAK_PGRB2_SPEC JJOB script

- Change NWROOT to PACKAGEROOT.
- Update PDY syntax.
- Update compath.py format.
- Update DATA default.
- Change RUN setting and add RUN2 setting.

Refs #419

KateFriedman-NOAA committed:
General WCOSS2 updates to JJOB scripts

- Change NWROOT to PACKAGEROOT.
- Update PDY syntax.
- Update compath.py format.
- Remove RUN_ENVIR=nco blocks.
- Change JGLOBAL_ATMOS_ANALYSIS_CALC to source "analcalc" instead of
"anal" from env file.
- Update DATA default.
- Remove references to jlogfile.

Refs #419

KateFriedman-NOAA committed:
Update config.efcs to run serial netcdf everywhere

- Remove machine case checks around OUTPUT_FILETYPE_ATM and
OUTPUT_FILETYPE_SFC settings.
- Set both variables to "netcdf" (serial netcdf) everywhere now.

Refs #419

KateFriedman-NOAA committed:
Whitespace cleanup in config.anal

Refs #419

KateFriedman-NOAA committed:
Add nth_echgres_gfs setting to config.analcalc

- When CDUMP=gfs set nth_echgres to nth_echgres_gfs

Refs #419

KateFriedman-NOAA committed:
Remove GEMPAKSH in config.gempak

- Remove setting of GEMPAKSH in config.gempak; JJOB now invoked directly
in rocoto job script gempak.sh.

Refs #419

KateFriedman-NOAA committed:
Fit2Obs, GSI monitoring, Tracker updates in config.vrfy

- Update Fit2Obs section to use new newm.1.5 version and add WCOSS2.
- Update Tracker section to use new v1.1.15.5 everywhere.
- Add COM_IN=$ROTDIR in GSI monitoring section; resolve unbound variable
error at runtime.

Refs #419

KateFriedman-NOAA committed:
WCOSS2 updates in config.base copies

- Change NWROOT to PACKAGEROOT.
- Update OBSPROC package settings. Temporarily use obsproc.v1.0.2 while
awaiting obsproc.v1.1.0 installs everywhere.
- Set ROTDIR to use compath.py in config.base.nco.static.

Refs #419

KateFriedman-NOAA committed:
Add GFS ops and WCOSS2 values into NCO config.fv3

Refs #419

KateFriedman-NOAA committed:
Updates to WCOSS2.env

- Add missing job sections into new WCOSS2.env.
- Add new settings added to develop into WCOSS2.env.
- Some additional updates to some job env settings.

Refs #419

KateFriedman-NOAA committed:
Change temp ncdiag MODULEPATH to append in module_base

- Change the temporary MODULEPATH para install settings in
module_base.wcoss2.lua for ncdiag to be an append instead of a prepend.
- When prepended the wrong path to a parm file from the g2tmpl module is
used. When appended the correct parm file from the g2tmpl module is used
because it uses the prod path first when loading the module.

Refs #419

KateFriedman-NOAA committed:
Correct PACKAGEROOT misspelling in config.base.emc.dyn

Refs #419

KateFriedman-NOAA committed:
Replace NWPROD with PACKAGEROOT in forecast_predet.sh

Refs #419

KateFriedman-NOAA committed:
Update WGRIB2 default in ush/scale_dec.sh

- WGRIB2 is no longer a package on disk and is a module.
- Set WGRIB2 variable default to use module wgrib2_ROOT bin path.

Refs #419

KateFriedman-NOAA committed:
Change WCOSS2.env from ksh to bash

Refs #419

KateFriedman-NOAA committed:
Update WGRIB2 default in ush scripts to use module

- Update the WGRIB2 setting default in ush scripts to no longer use
NWPROD and instead now use module wgrib2_ROOT bin path for wgrib2 exec.

Refs #419

KateFriedman-NOAA committed:
Replace NWPROD with PACKAGEROOT or module paths

- Remove commented out NWPROD setting in scripts/run_reg2grb2.sh.
- Replace NWPROD with PACKAGEROOT in ush scripts.
- Remove NWPROD in ush script utility default settings with module ROOT
paths.

Refs #419

KateFriedman-NOAA committed:
Update CFP settings for gldas job on WCOSS2

- Add USE_CFP and CFP_MP settings into gldas block in WCOSS2.env.
- When on WCOSS2 the gldas job should use CFP (USE_CFP=YES) but not use
the line counter value in the generated cfile (CFP_MP=NO).

Related to work happening in the GLDAS script absorption issue #1014.

Refs #419

KateFriedman-NOAA committed:
Cleanup ush/global_extrkr.sh

- Remove unneeded commented out lines and NWPROD duplicate lines.
- Replace NWPROD with PACKAGEROOT.

Refs #419

KateFriedman-NOAA committed:
Updates to ush/global_nceppost.sh

- Some cleanup.
- Replace NWPROD with HOMEgfs.

Refs #419

KateFriedman-NOAA committed:
Revert DATA default value in some JJOBS

- Change the default for DATA in some downstream job JJOB scripts to use
jobid again instead of PBS_JOBNAME.PBS_JOBID.
- The jobid variable is set in env files and uses PBS_JOBNAME.PBS_JOBID
in WCOSS2.env so it's unneeded to change in JJOBS.

Refs #419

KateFriedman-NOAA committed:
Update COMINsyn setting in configs

- Add COMINsyn to config.base.emc.dyn and have it use COMROOT and
gfs_ver syndat path. Set default for gfs_ver for now and remove when
version files are added later.
- Update default of COMINsyn in config.prep and config.vrfy.

Refs #419

KateFriedman-NOAA committed:
Add supported_resolutions to wcoss2.yaml host file

Refs #419

KateFriedman-NOAA committed:
Remove unused ush/global_extrkr.sh

Refs #419

KateFriedman-NOAA committed:
Add analcalc to Hera and Orion env files

- Similar to a change made in WCOSS2.env, update the if-check for
the "anal" job in env files for Hera and Orion to include "analcalc".
- The analcalc job needs to source the same settings as the anal job.

Refs #419

KateFriedman-NOAA committed:
Merge branch 'feature/dev-wcoss2' of https://github.com/KateFriedman-NOAA/global-workflow into feature/dev-wcoss2

* 'feature/dev-wcoss2' of https://github.com/KateFriedman-NOAA/global-workflow:
  Remove unused ush/global_extrkr.sh
  Add supported_resolutions to wcoss2.yaml host file
  Update COMINsyn setting in configs
  Revert DATA default value in some JJOBS

KateFriedman-NOAA committed:
Merge branch 'NOAA-EMC:develop' into feature/dev-wcoss2

KateFriedman-NOAA committed:
Merge branch 'NOAA-EMC:develop' into feature/dev-wcoss2

KateFriedman-NOAA committed:
Reverting Fit2Obs updates - will do separately

Refs #419

KateFriedman-NOAA committed:
Update config.resources for analcalc

Add needed settings into analcalc block of config.resources:
- add "ntasks=$npe_analcalc"
- add echgres variables: nth_echgres, nth_echgres_gfs

Refs #419

KateFriedman-NOAA commented:
Tests on Hera and Orion have both now completed. Atmos-only cycled and forecast-only coupled tests using `feature/dev-wcoss2` all reproduced controls using `develop`.

WalterKolczynski-NOAA commented on workflow/hosts/hera.yaml:
```diff
@@ -1,8 +1,8 @@
-nwprod: '/scratch1/NCEPDEV/global/glopara/nwpara'
-comroot: '/scratch1/NCEPDEV/global/glopara/com'
+packageroot: '/scratch1/NCEPDEV/global/glopara/nwpara'
+comroot: '/scratch1/NCEPDEV/rstprod/prod/com'
```
Why is comroot changed on hera?

KateFriedman-NOAA replied:
I have reverted it back to the prior setting. I had changed it to the new prod mirror location (/scratch1/NCEPDEV/rstprod/prod/com) so the tracker could get the syndat files but NCO won't add syndat to the mirror until next year (put in request last week).

Instead, I have:
1. Updated the existing syndat folder under /scratch1/NCEPDEV/global/glopara/com on Hera and /work/noaa/global/glopara/com/gfs/prod/syndat on Orion with the current syndat from ops on WCOSS2.
2. Added `cominsyn` paths to the hosts files.
3. Updated the setup scripts and `config.base.emc.dyn` to set `COMINsyn` using `cominsyn` from the hosts files.

WalterKolczynski-NOAA commented on util/sorc/webtitle.fd/makefile:
```diff
@@ -11,12 +11,12 @@ OBJS=   webtitle.o
 # PROFLIB	Library needed for profiling
 #
-FC =	ifort 
+FC =	ftn
```
The util directory is being removed by PR #1009. Any changes to these files need to be made as a PR to the [NOAA-EMC/gfs-util](https://github.com/NOAA-EMC/gfs-utils) repo

KateFriedman-NOAA replied:
Reverted, Will address in `gfs-util` repo as needed.

WalterKolczynski-NOAA commented on jobs/JGDAS_ATMOS_GEMPAK:
```diff
@@ -5,6 +5,24 @@ source "$HOMEgfs/ush/preamble.sh"
+echo
+echo "=============== BEGIN GEMPAK ==============="
+echo
+echo "=============== BEGIN TO SOURCE RELEVANT CONFIGS ==============="
```
The preamble announces the script beginning, it isn't needed here.

KateFriedman-NOAA replied:
I removed lines 8 and 9. Should lines 10 and 11 go as well?

WalterKolczynski-NOAA replied:
Probably, if only because other scripts don't do it.

KateFriedman-NOAA replied:
Done. I also removed the other "BEGIN TO SOURCE" announcement for the env file.

WalterKolczynski-NOAA commented on jobs/JGFS_ATMOS_GEMPAK:
```diff
@@ -5,13 +5,31 @@ source "$HOMEgfs/ush/preamble.sh"
 # GFS GEMPAK PRODUCT GENERATION
 ############################################
+echo
+echo "=============== BEGIN GEMPAK ==============="
```
Same here. The preamble announces the script start.

KateFriedman-NOAA replied:
Removed.

KateFriedman-NOAA replied:
I also removed the other "BEGIN TO SOURCE" announcement for the env file.

WalterKolczynski-NOAA commented on jobs/JGLOBAL_ATMOS_NCEPPOST:
```diff
@@ -70,7 +70,7 @@ export COMPONENT=${COMPONENT:-atmos}
 export APRUNP=${APRUN:-$APRUN_NP}
 export RERUN=${RERUN:-NO}
-export HOMECRTM=${HOMECRTM:-${NWROOT:-}/lib/crtm/${crtm_ver:-}}
+export HOMECRTM=${HOMECRTM:-${PACKAGEROOT:-}/lib/crtm/${crtm_ver:-}}
```
With empty default values for PACKAGEROOT and crtm_ver here, this line will produce incorrect paths but not error.

KateFriedman-NOAA replied:
Good catch. Should we fix the empty defaults in this PR or your task to remove "NCEP" from the post script names?

WalterKolczynski-NOAA replied:
Here. Just throw in some question marks if you need to.

KateFriedman-NOAA replied:
Done!

WalterKolczynski-NOAA commented on parm/config/config.base.emc.dyn:
```diff
@@ -76,9 +77,8 @@ export MODE="@MODE@" # cycled/forecast-only
-export HOMEobsproc_prep="$BASE_GIT/obsproc/obsproc_prep.v5.5.0_hpc-stack"
-export HOMEobsproc_network="$BASE_GIT/obsproc/obsproc_global.v3.4.2_hpc-stack"
-export HOMEobsproc_global=$HOMEobsproc_network
+#export HOMEobsproc="$BASE_GIT/obsproc/v1.1.0"
```
Remove dead code. Why are we not using the v1.1.0?

KateFriedman-NOAA replied:
I have reverted this back to the prior HOMEobsproc* variables. Need additional updates for the new prepobs modulefile so I will update obsproc/prepobs in a separate PR.

KateFriedman-NOAA replied:
Also, obsproc.v1.1.0 is coming with GFSv16.3, we're at v1.0.2 for GFSv16.2. Waiting for final tag of v1.1.0, may update to v1.0.2 in `develop` first. The prepobs version is v1.0.1 in both GFS versions.

WalterKolczynski-NOAA replied:
Okay, I just saw the nco version of this file using 1.1.0, so I was wondering why the difference.

KateFriedman-NOAA replied:
That was me getting ahead of things to set the incoming ops version in the nco version of that config. I'll redo the nco config obsproc update when I do the dev side change now instead.

WalterKolczynski-NOAA commented on parm/config/config.resources.nco.static:
```diff
@@ -0,0 +1,354 @@
+#!/bin/ksh -x
```
I know this is copied from ops branch, but we've already converted all the others to bash.

KateFriedman-NOAA replied:
Converted to bash. Thanks for catching!

KateFriedman-NOAA committed:
Back out obsproc updates, do in separate PR

Refs #419

KateFriedman-NOAA committed:
Updates for COMINsyn in supported platforms

- Change COMROOT/comroot back to prior path on Hera.
- Add cominsyn to setup_expt.py and hosts files to set COMINsyn
in config.base[.emc.dyn] on supported platforms.
- When syndat is added to ops mirror to Hera change comroot to
new mirror location to get syndat.

Refs #419

KateFriedman-NOAA committed:
Replace COMINsyn in config.base.emc.dyn with parsed value

Refs #419

KateFriedman-NOAA committed:
Change new config.resources.nco.static from ksh to bash

Refs #419

KateFriedman-NOAA committed:
Remove unneeded echos in GEMPAK JJOBS

Refs #419

KateFriedman-NOAA committed:
Fix empty defaults in jobs/JGLOBAL_ATMOS_NCEPPOST

- Change default for PACKAGEROOT and crtm_ver to ?.

Refs #419

KateFriedman-NOAA committed:
Remove other unneeded echo announcements in GEMPAK JJOBS

Refs #419

KateFriedman-NOAA committed:
Revert FC change in webtitle makefile

- Will make change to code makefile in new utils repo as needed.

Refs #419

KateFriedman-NOAA committed:
Merge branch 'NOAA-EMC:develop' into develop

KateFriedman-NOAA committed:
Merge branch 'develop' into feature/dev-wcoss2

* develop:
  Fix radiance verification failing to find diag files (#1031)
  Supported resolutions on platforms and defaults for mode (#1026)

KateFriedman-NOAA commented:
Sync merged with `develop` and resolved conflict in `workflow/hosts/orion.yaml`.

KateFriedman-NOAA commented:
@WalterKolczynski-NOAA I believe I have addressed your comments. Let me know if not or if you have any others. Thanks!

KateFriedman-NOAA committed:
Merge branch 'NOAA-EMC:develop' into feature/dev-wcoss2

KateFriedman-NOAA committed:
Retire base_svn from wcoss2 hostfile

Refs #1035 #419

KateFriedman-NOAA committed:
Merge branch 'feature/dev-wcoss2' of https://github.com/KateFriedman-NOAA/global-workflow into feature/dev-wcoss2

* 'feature/dev-wcoss2' of https://github.com/KateFriedman-NOAA/global-workflow:
  initial commit for incoming yaml work (#1029)

KateFriedman-NOAA committed:
Backout prepobs module load addition

- New obsproc/prepobs changes will happen in separate PR

Refs #419

KateFriedman-NOAA committed:
Merge branch 'NOAA-EMC:develop' into feature/dev-wcoss2

KateFriedman-NOAA committed:
Put prepobs module load back in module_base.wcoss2

It was already in develop, put it back.

Refs #419

KateFriedman-NOAA committed:
Merge branch 'feature/dev-wcoss2' of https://github.com/KateFriedman-NOAA/global-workflow into feature/dev-wcoss2

* 'feature/dev-wcoss2' of https://github.com/KateFriedman-NOAA/global-workflow:
  Remove base_svn from Hera and Orion hosts files (#1036)
